### PR TITLE
Fixes #30934 - improve template invocation_form inputs

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Using `react_form_input` helper, that was introduced in 810504a
for rendering template invocation react inputs.

Hides all the input type implementations into a one helper method
to hide the implementation differences between React and Rails implemented inputs.

### CODE REVIEW

File: app/helpers/templates_helper.rb [hunk 1]
```diff
@@ -58,27 +58,29 @@ def hide_resource_type_input(obj)
...
-    })
+    options.reverse_merge!(label: input.name, id: input.name, label_help: input.description, required: input.required)
+
+    if input.options.present?
+      selectable_f(f, :value, input.options_array, { include_blank: !input.required }, id: input.name, onchange: options[:onchange])
+    elsif input.value_type == 'plain' || input.value_type.nil?
+      textarea_f(f, :value, rows: 2, onchange: options[:onchange], id: input.name, class: input.hidden_value? ? 'masked-input' : '')
+    else
+      input_type = input.value_type
+      input_type = 'dateTime' if input_type == 'date'
```
Comment (ezr-ondrej) [inline app/helpers/templates_helper.rb]: I've dropped the input value type renaming to separate PR (#8069) so I'm doing this instead, let's do it one by one :+1:

**Comment by ezr-ondrej** [comment]
I think this is ready :)

File: app/helpers/templates_helper.rb [hunk 2]
```diff
@@ -58,27 +58,27 @@ def hide_resource_type_input(obj)
...
+      selectable_f(f, :value, input.options_array, { include_blank: !input.required }, options)
+    elsif input.value_type == 'plain' || input.value_type.nil?
+      textarea_f(f, :value, options.merge(rows: 2, class: input.hidden_value? ? 'masked-input' : ''))
+    else
+      input_type = input.value_type
+      input_type = 'dateTime' if input_type == 'date'
+      if input_type == 'search'
+        input_type = 'autocomplete'
+        resource_type = input.resource_type&.tableize
+        options.merge({ resource_type: resource_type,
```
Comment (MariaAga) [inline app/helpers/templates_helper.rb]: This should be `merge!`  as currently the search options aren't added

Comment (ezr-ondrej) [reply app/helpers/templates_helper.rb]: Thanks! Good catch!

**Comment by MariaAga** [comment]
In React the default description for date was `'Format is yyyy-mm-dd HH-mm-ss'` can you add it here? Other than that it works great!

**Comment by ezr-ondrej** [comment]
> In React the default description for date was `'Format is yyyy-mm-dd HH-mm-ss'` can you add it here? Other than that it works great!

I didn't feel it's necessary to have while we have the datetime select, but I've added it in separate commit the same way it was added in React before.

**Comment by MariaAga** [comment]
I'm not sure why but it's still shows up empty for me. But you're right since we have the datetime select it's not necessary

**Comment by ezr-ondrej** [comment]
> I'm not sure why but it's still shows up empty for me. But you're right since we have the datetime select it's not necessary

Right, the description could have been empty string and in that case `||= ` operator would not work there, I've added the `.presence` it should fix the issue :+1:

File: app/helpers/templates_helper.rb [hunk 3]
```diff
@@ -58,27 +58,27 @@ def hide_resource_type_input(obj)
...
+
+    if input.options.present?
+      selectable_f(f, :value, input.options_array, { include_blank: !input.required }, options)
+    elsif input.value_type == 'plain' || input.value_type.nil?
+      textarea_f(f, :value, options.merge(rows: 2, class: input.hidden_value? ? 'masked-input' : ''))
+    else
+      input_type = input.value_type
+      if input_type == 'date'
+        input_type = 'dateTime'
+        options[:label_help] ||= 'Format is yyyy-mm-dd HH-mm-ss'
```
Comment (adamruzicka) [inline app/helpers/templates_helper.rb]: We should distinguish between mm for month and MM for minutes

File: app/helpers/templates_helper.rb [hunk 4]
```diff
@@ -58,27 +58,27 @@ def hide_resource_type_input(obj)
...
+      input_type = input.value_type
+      if input_type == 'date'
+        input_type = 'dateTime'
+        options[:label_help] ||= 'Format is yyyy-MM-dd HH-mm-ss'
+      elsif input_type == 'search'
+        input_type = 'autocomplete'
+        resource_type = input.resource_type&.tableize
+        options.merge!(resource_type: resource_type, use_key_shortcuts: false, url: search_path(resource_type))
+      end
+      react_form_input(input_type, f, :value, options)
```
Comment (adamruzicka) [inline app/helpers/templates_helper.rb:81]: If I have a rex template with date input it works fine. It gets sent, everything is in the right place, but if I rerun the job, the value from the last invocation doesn't get prefilled. It works fine for other types. It is possible it is not related to this change at all, just saying.

**Review by adamruzicka — COMMENTED** [review]
Seems to work fine apart from pre-filling of date inputs

File: app/helpers/templates_helper.rb (hunk 4 above)
Comment (ezr-ondrej) [reply app/helpers/templates_helper.rb:81]: Nice catch, altough it's not an issue of this PR, but rather the DatePicker itself, I've opened another PR to fix that: #8122

**Comment by adamruzicka** [comment]
Needs a rebase

**Comment by ezr-ondrej** [comment]
Rebased :)

**Review by adamruzicka — APPROVED** [review]
Re-tested again, works nicely

**Comment by adamruzicka** [comment]
Thank you @ezr-ondrej !

**Comment by tbrisker** [comment]
2.3 - 7187d183c

**Pull request merged**